### ISSUE TITLE
support for xhamsterlive

### ISSUE THREAD

**Issue by kusti420** [ytdl-org/youtube-dl#30792]
xhamsterlive.com

## Checklist

- [x] I'm reporting a new site support request
- [x] I've verified that I'm running youtube-dl version **2021.12.17**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that none of provided URLs violate any copyrights
- [x] I've searched the bugtracker for similar site support requests including closed ones

## Example URLs

- Single stream: [replace with any stream thats live there right now ( https://xhamsterlive.com/Akilla_Queen ) for example]

## Description

ERROR: Unsupported URL: https://xhamsterlive.com/Akilla_Queen is always raised. 
saw someone else request the same thing few years ago, but this site still isnt supported and that issue has been closed:
https://github.com/ytdl-org/youtube-dl/issues/21088

**Comment by afterdelight** [comment]
try streamlink

**Comment by kusti420** [comment]
> 

i just tried it out but it doesnt seem to work with it, says that no plugin can handle that url and i didnt find any plugins for xhamsterlive either

**Comment by afterdelight** [comment]
Well you just have to extract the .m3u8 index yourself then download it with streamlink or yt-dl or you-get

**Comment by kusti420** [comment]
coming back to it. ffmpeg is able to download it in a following way for example: `ffmpeg -headers "User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/123.0.0.0 Safari/537.36 Accept: */* Accept-Language: en-US,en;q=0.9 Accept-Encoding: gzip, deflate, br Origin: https://xhamsterlive.com Referer: https://xhamsterlive.com/ Connection: keep-alive" -i "https://media-hls.doppiocdn.com/b-hls-12/162907341/162907341.m3u8?playlistType=lowLatency" -c copy -bsf:a aac_adtstoasc -movflags +faststart+frag_keyframe+empty_moov -timeout 10 -reconnect 1 -reconnect_at_eof 1 -reconnect_streamed 1 -reconnect_delay_max 10 -reconnect_on_network_error 1 -reconnect_on_http_error 1 "stream.mp4"`

so basically it downloads from xhamsterlives media server `https://media-hls.doppiocdn.com/b-hls-{magic_number?}/{user_id}/{user_id}.m3u8?playlistType=lowLatency`

user_id can be requested from `https://xhamsterlive.com/api/front/models/username/{username}`

in my examples case i cant figure out where "b-hls-12" is coming from, this value varies from user to user and stream to stream.

**Comment by dirkf** [comment]
In the page's hydration JSON, there is (`\u002F` = `/`)
```py
"hlsStreamUrlTemplate":"https:\u002F\u002Fedge-hls.{cdnHost}\u002Fhls\u002F{streamName}\u002Fmaster\u002F{streamName}{suffix}.m3u8",
```
`streamName` is present and appears to be the `id` of the model. There's also a list of `cdnHost`s at `"hlsStreamHosts"`. As to `suffix`, more research needed.
